source/slang/slang-parser.cpp

```diff
@@ -4364,6 +4387,44 @@ namespace Slang
         return attrDecl;
     }
 
+    static void addSpecialGLSLModifiersBasedOnType(
+        Parser* parser,
+        Decl* decl,
+        Modifiers* modifiers)
+    {
+        // we intentionally have findModifier twice since GLSLOffsetLayoutAttribute is rare
+        // better to early leave
+        auto bindingMod = modifiers->findModifier<GLSLBindingAttribute>();
+        if (!bindingMod) return;
+        auto varDeclBase = as<VarDeclBase>(decl);
+        if (!varDeclBase) return;
+        auto declRefExpr = as<DeclRefExpr>(varDeclBase->type.exp);
+        if (!declRefExpr) return;
```

Comment:
The first comment gave me the impression that "findModifer()" is an expensive function call and we are trying to avoid calling it.
Why don't we avoid the first call of "findModifier()" by checking the other variables first?
If "varDeclBase" is a null_ptr, we didn't need to call "findModifier()".

Reply:
the order of the operations should be moved, I agree, I will remove the comment (not really important).